source/lex.tex

```diff
@@ -660,31 +660,36 @@
 \end{bnf}
 
 \pnum
+The sequences in both forms of \grammarterm{header-name}{s} are mapped in an
+\impldef{mapping header name to header or external source file} manner to headers or to
+external source file names as specified in~\ref{cpp.include}.
 \begin{note}
 Header name preprocessing tokens only appear within
 a \tcode{\#include} preprocessing directive,
 a \tcode{__has_include} preprocessing expression, or
 after certain occurrences of an \tcode{import} token
 (see~\ref{lex.pptoken}).
 \end{note}
-The sequences in both forms of \grammarterm{header-name}{s} are mapped in an
-\impldef{mapping header name to header or external source file} manner to headers or to
-external source file names as specified in~\ref{cpp.include}.
 
 \pnum
-The appearance of either of the characters \tcode{'} or \tcode{\textbackslash} or of
-either of the character sequences \tcode{/*} or \tcode{//} in a
-\grammarterm{q-char-sequence} or an \grammarterm{h-char-sequence}
-is conditionally-supported with \impldef{meaning of \tcode{'}, \tcode{\textbackslash},
-\tcode{/*}, or \tcode{//} in a \grammarterm{q-char-sequence} or an
-\grammarterm{h-char-sequence}} semantics, as is the appearance of the character
-\tcode{"} in an \grammarterm{h-char-sequence}.
-\begin{footnote}
+The appearance of either of the characters \unicode{0027}{apostrophe},
+\unicode{005c}{reverse solidus}, or of either of the character sequences
+\tcode{/*} or \tcode{//} in a \grammarterm{q-char-sequence} is
+conditionally-supported with
+\impldef{meaning in a \grammarterm{q-char-sequence} of \tcode{'},
+\tcode{\textbackslash}, \tcode{/*}, or \tcode{//}} semantics.
+The appearance of any of the characters \unicode{0022}{quotation mark},
+\unicode{0027}{apostrophe}, \unicode{005c}{reverse solidus}, or of either of the
+character sequences \tcode{/*} or \tcode{//} in a \grammarterm{h-char-sequence}
+is conditionally-supported with
+\impldef{meaning in a \grammarterm{h-char-sequence} of \tcode{""}, \tcode{'},
+\tcode{\textbackslash}, \tcode{/*}, or \tcode{//}} semantics.
```

Comment:
I like the \unicode stuff, I like the reordering, I like the footnote -> note change.

I don't like repeating the list of characters.  Can we just extend the original \impldef to say "... in a q-char-sequence or h-char-sequence" ?  Yes, we omit " that way, but the text in the impldef annex doesn't need to be comprehensive.  Or maybe say "meaning of certain characters in a q-char-sequence or h-char-sequence" to cover everything?

Reply:
I prefer this duplication, as it will allow me to file a CWG issue that neither of the two comment forms can appear in a _h-char-sequence_ as they will be processed as comments, unlike in the _q-char-sequence_.

Reply:
> I prefer this duplication, as it will allow me to file a CWG issue that ...

No amount of editorial refactoring can affect the validity of a CWG issue submission. Either there is a CWG-level problem, or there isn't, or the supposedly-editorial refactoring isn't actually editorial.

Reply:
Yes, my edit above is purely editorial.  However, the CWG issue I want to file will make it clear that the sets of characters have even less of an overlap, as the two comment-openers are an issue for only _h-char_sequences_ and the embedded quote is a problem for only _q-char-sequences_, and the overlap is just single-quote and backslash.

I can file the CWG issue without this refactoring, but it would look much simpler with it, and I believe the refactored version that removes the comment-openers from _h-char-sequence_ consideration would also be our preferred final outcome.

I can switch back to a common sentence form is that remains your preferred rendering before I file the CWG issue though.

Reply:
May I suggest to limit this editorial pull request to these non-contentious items:

> I like the \unicode stuff, I like the reordering, I like the footnote -> note change.

After the merge, you'll post your CWG issue, and its suggested resolution (by happenstance) can disentangle the sentence to give \impldef some joy.

Reply:
Posted the more minimal edit, and changed the commit message accordingly.  Updated the description at the top of this PR too.